### ISSUE TITLE
Langflow Python API delayed response 

### ISSUE THREAD

**Issue by sampratha28** [langflow-ai/langflow#1287]
I am callling the langflow python API through python requests module.

it is taking too long to respond compared to the time taken for response in the langflow UI.

How do i fix this

**Comment by ogabrielluiz** [comment]
Hi @sampratha28 

When you call the API the build process has to occur while in the UI you have to press the build button before starting the chat. The first message always builds the flow and then that session is cached so you can reuse it.

**Comment by sampratha28** [comment]
@ogabrielluiz @yamonkjd thanks. This was helpful

But for how long will the session be cached? Because after a certain time, i am getting 'graph not built' error. But it works the next time i hit the API with the same session id. 
Is there any timeout happening there?

**Comment by yamonkjd** [comment]
> @ogabrielluiz @yamonkjd thanks. This was helpful
> 
> But for how long will the session be cached? Because after a certain time, i am getting ''Graph not found in the session' error. But it works the next time i hit the API with the same session id. Is there any timeout happening there?

I don't know if I'm judging correctly, but it seems that when there are more than two workers, the session may not be found properly in certain situations.

**Comment by ogabrielluiz** [comment]
That is correct when not using a cache like Redis. The session data is store in memory and can't be shared between the two processes.

**Comment by anovazzi1** [comment]
Hello, @YamonBot 
Sorry for the delay. Did you try using the new version? Does the error still persist?

**Comment by carlosrcoelho** [comment]
Hi @sampratha28 

We hope you're doing well. Just a friendly reminder that if we do not hear back from you within the next 3 days, we will close this issue. If you need more time or further assistance, please let us know. 

Thank you for your understanding!

**Comment by carlosrcoelho** [comment]
Thank you for your contribution! This issue will be closed. If you have any questions or encounter another problem, please open a new issue and we will be ready to assist you.
